core/txpool/legacypool/list.go

```diff
@@ -162,9 +169,9 @@ func (m *SortedMap) Cap(threshold int) types.Transactions {
 	// Otherwise gather and drop the highest nonce'd transactions
 	var drops types.Transactions
 	slices.Sort(*m.index)
-	for size := len(m.items); size > threshold; size-- {
-		drops = append(drops, m.items[(*m.index)[size-1]])
-		delete(m.items, (*m.index)[size-1])
+	for _, nonce := range (*m.index)[threshold:] {
```

Comment:
they are essentially equivalent, with no performance difference.

i would prefer to keep it as now.